src/OrbitGl/App.cpp

```diff
@@ -347,6 +347,8 @@ void OrbitApp::OnCaptureStarted(const orbit_grpc_protos::CaptureStarted& capture
         // this task is completely executed.
         capture_data_ = std::make_unique<CaptureData>(
             capture_started, file_path, std::move(frame_track_function_ids), data_source_);
+        capture_data_->set_memory_warning_threshold_kb(
```

Comment:
I dont understand, we are saving it twice? (CaptureData and DataManager)? Is that something we were doing before?

Reply:
We use `DataManager` for the options for the next capture and we set the memory warning threshold in `DataManager` when the user changes it in  the capture options dialog (see  `OrbitMainWindow::on_actionCaptureOptions_triggered` for more details). 
At this time, the capture data may not be created yet.

Also, we are doing this for `memory_sampling_period`. You didn't see a set method in capture data because the memory sampling period is set in the capture data constructor.